src/api/dataops/queries/insertQuery.spec.ts

```diff
@@ -1,51 +1,48 @@
-// tslint:disable:no-string-literal
+import * as faker from "faker";
 import { createRequestExecuterMock } from "../../../../spec/testUtils";
 import { InsertQuery } from "./insertQuery";
 
 describe("InsertQuery class", () => {
   let projectID: string;
   let dataset: string;
+  let qe;
+  let subject: InsertQuery<any, any>;
+
+  const fakeRecord = {
+    title: faker.random.words(3),
+    user_id: faker.random.uuid(),
+  };
 
   beforeAll(() => {
     dataset = "dataset";
     projectID = "projectID";
   });
 
+  beforeEach(() => {
+    qe = createRequestExecuterMock(projectID, dataset);
+    subject = new InsertQuery(qe, [fakeRecord], dataset);
+  });
+
   describe("when instantiating a insertQuery object directly", () => {
     it("should be able to return required object", (done) => {
-        let qe = createRequestExecuterMock(projectID, dataset);
-        let query = new InsertQuery(qe, [{title: "Another first post", user_id: 1}], dataset);
-        expect(query).toBeDefined();
-        done();
+      expect(subject).toBeDefined();
+      done();
     });
   });
 
   describe("when instantiating a insertQuery object from client", () => {
     it("should be able to invoke methods exposed by it", (done) => {
-        let qe = createRequestExecuterMock(projectID, dataset);
-        let query = new InsertQuery(qe, [{title: "Another first post", user_id: 1}], dataset);
-        expect(typeof query.execute).toBe("function");
-        done();
-    });
-  });
-
-  describe("when instantiating a insertQuery object from client", () => {
-    it("its query object should have desired properties", (done) => {
-        let qe = createRequestExecuterMock(projectID, dataset);
-        let queryObj: any = new InsertQuery(qe, [{title: "Another first post", user_id: 1}], dataset);
-        expect(queryObj).toBeDefined();
-        expect(queryObj.records).toEqual([{title: "Another first post", user_id: 1}]);
-        done();
+      expect(typeof subject.execute).toBe("function");
+      done();
```

Comment:
Do we really need to use done() here? Also, it's a pretty weird test as for me

Reply:
no, removed!